python/dolfinx/fem/linearvariationalsolver.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+import typing
+
+from petsc4py import PETSc
+from ufl.equation import Equation
+from dolfinx import fem
+
+
+class LinearVariationalSolver():
+    assemble_b = True
+
+    def __init__(self, variational_problem: Equation, u: fem.Function,
```

Comment:
We want to remove `Equation` - it's an abuse of syntax. Pass two forms instead.

I'm not sure whether or not we want to pass a solution function, or let the object create it.

Reply:
I agree with @garth-wells here.